File: cadwiki-nuget/cadwiki.DllReloader/AutoCAD/UiRibbon/Buttons/Creator.vb
Imports System.Drawing
Imports System.Reflection
Imports System.Windows.Media.Imaging
Imports Autodesk.Windows

Namespace AutoCAD.UiRibbon.Buttons
    Public Class Creator
        Public Shared Function Create(ByVal name As String,
                                 ByVal text As String,
                                 ByVal tooltip As String,
                                 ByVal bitMap As Bitmap,
                                 ByVal assemblyName As String,
                                 ByVal fullClassName As String,
                                 ByVal methodName As String,
                                 ByVal parameters As Object(),
                                 ByVal appDomainReloader As AutoCADAppDomainDllReloader,
                                 ByVal iExtensionAppAssembly As Assembly) As RibbonButton


            Dim ribbonButton = New RibbonButton()
            ribbonButton.Name = name
            ribbonButton.ShowText = True
            ribbonButton.Text = text
            ribbonButton.Size = RibbonItemSize.Standard

            If bitMap IsNot Nothing Then
                Dim image As BitmapSource = NetUtils.Bitmaps.CreateBitmapSourceFromBitmap(bitMap)
                If image IsNot Nothing Then
                    ribbonButton.Image = image
                    ribbonButton.ShowImage = True
                End If
            End If

            Dim uiRouter As UiRouter = New UiRouter(assemblyName, fullClassName, methodName, parameters, appDomainReloader, iExtensionAppAssembly)
            ribbonButton.CommandParameter = uiRouter
            ribbonButton.CommandHandler = New GenericClickCommandHandler()
            ribbonButton.ToolTip = tooltip
            Return ribbonButton
        End Function
    End Class

End Namespace

File: cadwiki-nuget/cadwiki.DllReloader/AutoCAD/UiRibbon/Buttons/Creator.cs
using System;
using System.Drawing;
using System.IO;
using System.Reflection;
using Autodesk.Windows;

namespace cadwiki.DllReloader.AutoCAD.UiRibbon.Buttons
{
    public class Creator
    {
        public static RibbonButton Create(string name, string text, string tooltip, Bitmap bitMap, string assemblyName, string fullClassName, string methodName, object[] parameters, AutoCADAppDomainDllReloader appDomainReloader, Assembly iExtensionAppAssembly)
        {


            var ribbonButton = new RibbonButton();
            ribbonButton.Name = name;
            ribbonButton.ShowText = true;
            ribbonButton.Text = text;
            ribbonButton.Size = RibbonItemSize.Standard;

            if (bitMap is not null)
            {
                try
                {
                    var image = NetUtils.Bitmaps.CreateBitmapSourceFromBitmap(bitMap);
                    if (image is not null)
                    {
                        ribbonButton.Image = image;
                        ribbonButton.ShowImage = true;
                    }
                }
                catch (Exception ex) when (
                    ex is TypeInitializationException ||
                    ex is FileNotFoundException ||
                    ex is DllNotFoundException ||
                    ex is TypeLoadException ||
                    ex is Exception)
                {
                    // Icon loading failed — button renders without an icon.
                    System.Diagnostics.Debug.WriteLine(
                        $"[Creator] Warning: Icon load failed for button '{name}': {ex.Message}");
                }
            }

            var uiRouter = new UiRouter(assemblyName, fullClassName, methodName, parameters, appDomainReloader, iExtensionAppAssembly);
            ribbonButton.CommandParameter = uiRouter;
            ribbonButton.CommandHandler = new GenericClickCommandHandler();
            ribbonButton.ToolTip = tooltip;
            return ribbonButton;
        }
    }

}
